middleware/check-auth.js

```diff
@@ -0,0 +1,16 @@
+const jwt = require('jsonwebtoken');
+
+const checkAuth = async (req, res, next) => {
+    try{
+        const token = req.headers.authorization.split(" ")[1];
+        const decodedToken = jwt.verify(token, process.env.JWT_KEY);
```

Comment:
Please check that `authorization` is defined in the payload before proceeding. If not, you want to throw an error to tell users to sign in first.

```js
        const auth = req.headers.authorization
        if (!auth) throw new Error('You need to sign in')
        
        const token = auth.split(" ")[1];
        const decodedToken = jwt.verify(token, JWT_KEY);
```